Improve dedent (fixes #1202)

Cito opened the pull request:
As discussed in #1202, this changes the behavior of the dedent function that is used in unit tests to not return raw strings any more. Also added tests for the function itself.

leebyron commented on src/language/__tests__/printer-test.js:
```diff
@@ -190,7 +190,7 @@ describe('Printer: Query document', () => {
       fragment frag on Friend {
         foo(size: $size, bar: $b, obj: {key: "value", block: """
-          block string uses \"""
+          block string uses \\"""
```
Instead of changing this to `\\`, could you use `String.raw` as discussed in #1202? This test case intends to show the actual GraphQL string written, so avoiding double-encoding is intentional.

leebyron commented on src/utilities/__tests__/schemaPrinter-test.js:
```diff
@@ -212,7 +212,7 @@ describe('Type System Printer', () => {
 
       type Root {
-        singleField(argOne: String = "tes\t de\fault"): String
+        singleField(argOne: String = "tes\\t de\\fault"): String
```
same

Cito commented:
That's true, using String.raw is clearer for the printer tests. I changed it and added another unit test to make sure interpolation also works properly when dedent is used as a string tag.

IvanGoncharov commented on src/jsutils/__tests__/dedent-test.js:
```diff
@@ -0,0 +1,113 @@
+      `).to.equal(
+      'type Query {\n  me: User\n}\n\n' +
+        'type User {\n  id: ID\n  name: String\n}\n',
+    );
```
In the current form, these tests are hard to read.
Would be great to have test string in a separate variable and expected string in [readable form](https://github.com/graphql/graphql-js/blob/master/src/language/__tests__/blockStringValue-test.js#L14-L21), e.g.:
```js
const output = dedent`
  type Query {
    me: User
  }
`;

expect(output).to.equal([
  'type Query {',
  '  me: User',
  '}',
].join('\n'));

Cito replied:
Yes, that looks a bit nicer, particularly if you're using that style already elsewhere. I refactored it already.

Cito commented:
Rebased onto current master now.
